### ISSUE TITLE
ci: release process broken for 2.7.0

### ISSUE THREAD

**Issue by apricote** [hetznercloud/csi-driver#611]
### TL;DR

Release 2.7.0 did not publish successfully

### Expected behavior

Release to be published as expected

### Observed behavior

[Release pipeline failed](https://github.com/hetznercloud/csi-driver/actions/runs/9193285609/job/25283944049) in `./hack/release-publish-helm-chart.sh`:

### Log output

```
   • publishing
    • custom publisher
      • publishing                                   cmd=./hack/release-publish-helm-chart.sh artifact=csi-driver_2.7.0_checksums.txt
      • took: 1s
  • took: 1s
  ⨯ release failed after 1m24s               error=custom publisher: failed to publish artifacts: publishing: ./hack/release-publish-helm-chart.sh failed: exit status 128: Cloning into 'csi-driver-chart-repo.okPkp'...
fatal: active `post-checkout` hook found during `git clone`:
	/home/runner/work/csi-driver/csi-driver/csi-driver-chart-repo.okPkp/.git/hooks/post-checkout
For security reasons, this is disallowed by default.
If this is intentional and the hook should actually be run, please
run the command again with `GIT_CLONE_PROTECTION_ACTIVE=false`
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

### Additional information

This is related to https://github.com/git-lfs/git-lfs/issues/5749 and will be fixed with a new Git version. Once this new version is released and available on GitHub Actions, we can simply re-run the release pipeline.

I have marked the `v2.7.0` release as a pre-release for now and added a note to inform users about this.

**Comment by resmo** [comment]
missing docker images (tags?) for 2.7.0 e.g. hetznercloud/hcloud-csi-driver:v2.7.0


https://hub.docker.com/r/hetznercloud/hcloud-csi-driver/tags

**Comment by apricote** [comment]
Yes. The whole release failed and nothing was published. We are waiting for a new release of the GitHub Runner images right now, which will use Git 1.45.2 and fix the issue: https://groups.google.com/g/git-packagers/c/x6-nKLV20aE

**Comment by loxy** [comment]
x-swarm was published under docker hub. When to expect the other images?

**Comment by apricote** [comment]
We are currently waiting for the new runner-image with Git 1.45.2 to be released: https://github.com/actions/runner-images/pull/9988

**Comment by apricote** [comment]
2.7.0 is now released everywhere :) Sorry for the delay!
